comparison.py

```diff
@@ -1,29 +1,64 @@
 import numpy as np
 from hmc_sampler import LeapfrogHMC
 from hmc_sampler import U7HMC
+import gelman_rubin
 
+
+number_of_chains = 2
 chain_length = 5000
-#Länge der Schrittweite im Leapfrog
 stepsize = 1.5
-#Anzahl der Schritte im Leapfrog
 trajectory_length = 10
-
 dimension = 2
 
-x_0 = np.zeros(dimension)
+x_0 = np.zeros(dimension)   #may be chosen randomly for different chains
+
+
 def log_prob(x):
-    return -0.5* np.sum(x**2 )
-#gradient der log Dichte funktion
+
+    return -1./2 * np.sum(x**2)
+
+
 def gradient_log_prob(x):
-    return -1*x
+
+    return -1 * x
+
 
 def hessian_log_prog(x):
-    return -1 *np.identity(2)
 
-leapfrog_HMC = LeapfrogHMC(x_0, chain_length, log_prob, gradient_log_prob, stepsize, trajectory_length)
-chain, acceptance_rate = leapfrog_HMC.build_chain()
-print(chain[:10])
+    return -1 * np.identity(2)
+
+
+"""creating the chains for the different integrators respectively"""
+u7_HMCs = [U7HMC(log_prob, gradient_log_prob, hessian_log_prog,
+                 stepsize, trajectory_length) for i in range(number_of_chains)]
```

Comment:
```suggestion
u7_HMCs = [U7HMC(log_prob, gradient_log_prob, hessian_log_prog,
                 stepsize, trajectory_length) for _ in range(number_of_chains)]
```
In Python, it's somewhat standard practice to denote unneeded variables with and underscore `_` so as to not make the reader wonder what, in this case, the `i` might be for.